feat: non-local jumps (#4566)
JetBrains/kotlin-web-site: docs/topics/inline-functions.md

>
{style="note"}

> ## Non-local jump expressions

Reviewer: The documentation on non-local break/continue/return is duplicated in `inline-functions.md` ("Inline functions") and `returs.md` ("Returns and jumps")

I think, it'd be better to place the docs only to one of the pages. and another page could just have a link to the first one.

But if you think that duplication is not a problem, you can keep it "as is". It's up to you.

Reviewer: To my mind, "non-local jumps" is more of a feature of `inline` functions. And the `returns.md` could be renamed to smth like `jump-labels.md` and be minimized to only talk about labels

Author: In my mind, non-local feature was as important as the labels, so I created an equal structure for both of them in returns.md

But you're right, if it's just a specific feature of inline functions, we can minimize it in returns.md down to just a link to the main Inline functions page with detailed info and avoid duplication

}
```

### Break and continue

> This feature is currently [Experimental](components-stability.md#stability-levels-explained).
> We're planning to stabilize it in future releases.
> To opt in, use the `-Xnon-local-break-continue` compiler option.

> > We would appreciate your feedback on it in [YouTrack](http://kotl.in/issue).

Reviewer: It's not clear to me how much sense it makes to point to the YouTrack new issue page in such kind of sentence. At least, it's slightly unexpected.

Maybe a better option would be to point to the issue itself https://youtrack.jetbrains.com/issue/KT-1436

>
{style="warning"}

> Similar to non-local `return`, you can apply `break` and `continue` [jump expressions](returns.md) within lambdas passed
> as arguments to an inline function that encloses the loop:

Reviewer 2: ```suggestion
Similar to non-local `return`, you can apply `break` and `continue` [jump expressions](returns.md) in lambdas passed
as arguments to an inline function that encloses a loop:
```

```kotlin

> fun processList(elements: List&lt;Int&gt;): Boolean {

Author: No, it was necessary for the XML version of this block in What's new, here it's redundant, thank you for noticing!
